### ISSUE TITLE
find_dotenv doesn't work as expected

### ISSUE THREAD

**Issue by voidlily** [theskumar/python-dotenv#24]
# What I did

```python
from dotenv import find_dotenv, load_dotenv
load_dotenv(find_dotenv())
```

# What I expected

The `.env` file to be loaded properly

# What I got

The `.env` file was not found

# Additional information

I print-debugged the code and found the following path when it walks up the directory tree

```
/usr/local/lib/python3.5/site-packages/dotenv
/usr/local/lib/python3.5/site-packages
/usr/local/lib/python3.5
/usr/local/lib
/usr/local
/usr
/
```

This might be from the usage of `__file__` in that module.

Is this the intended behavior of `find_dotenv`?

**Comment by rulethr33** [comment]
That's not intended behavior and sure we can use a fix. I'll have a look
into it.

**Comment by theskumar** [comment]
That's not intended behavior and sure we can use a fix. I'll have a look into it. Thanks for reporting.

**Comment by williamsmj** [comment]
I see the same behaviour. Looks like find_dotenv() is broken. I tested this on both Python 2.7 and 3.5.

```
$ cat .env
TEST=1234
$ cat settings.py
from dotenv import find_dotenv
print(find_dotenv())
$ python settings.py

$
```

Based on my limited understanding, I _think_ you want ```path = os.path.abspath(inspect.stack()[-1][1])``` [here](https://github.com/theskumar/python-dotenv/blob/1fd40c5bcedc4f3f341061dd1dab075e659a669f/dotenv/main.py#L128).

**Comment by theskumar** [comment]
@voidlily @williamsmj I've added a fix for the .env finding logic. It now starts finding from the python file from where it's called, then move up the directory till `/`.

You can try it out by installing from latest master.

```
pip install --upgrade https://github.com/theskumar/python-dotenv/archive/master.zip
```

If things look, I'll make a new release on pypi.

Cheers,

**Comment by williamsmj** [comment]
Works for me 👍

```bash
$ cat .env
TEST=1234
$ cat settings.py
from dotenv import find_dotenv
print(find_dotenv())
$ python settings.py
/Users/mike/tmp/dotenv/.env
$```

**Comment by isms** [comment]
@voidlily good catch! @theskumar that one is on me -- oops :disappointed: 

Just tested again and it works for me.

**Comment by theskumar** [comment]
@isms no worries! 

I've made released v0.5.1 🎉  which contains this fix.
